cg/store/api/find_business_data.py

```diff
@@ -330,13 +344,94 @@ def get_number_of_reads_for_sample_passing_q30_threshold(
         return reads_count if reads_count else 0
 
+    def get_number_of_reads_for_sample_on_flow_cell_from_sample_lane_metrics(
+        self, sample_internal_id: str, flow_cell_name: str
+    ) -> int:
+        total_reads_query: Query = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[
+                SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME,
+                SequencingMetricsFilter.FILTER_TOTAL_READ_COUNT_FOR_SAMPLE,
+            ],
+            sample_internal_id=sample_internal_id,
+            flow_cell_name=flow_cell_name,
+        )
+        reads_count: Optional[int] = total_reads_query.scalar()
+        return reads_count if reads_count else 0
+
+    def get_samples_internal_ids_on_flow_cell_from_sample_lane_metrics(
+        self, flow_cell_name: str
+    ) -> List[str]:
+        """Get samples on a given flow cell."""
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=[SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME],
+            flow_cell_name=flow_cell_name,
+        ).all()
+
+        return list(set([metric.sample_internal_id for metric in metrics]))
+
+    def get_average_passing_q30_for_sample_from_sample_lane_metrics(
+        self, sample_internal_id: str, flow_cell_name: str
+    ) -> float:
+        """Get average q30 for a sample on a given flow cell."""
+        sample_lane_metrics_filter_functions = [
+            SequencingMetricsFilter.FILTER_BY_FLOW_CELL_NAME,
+            SequencingMetricsFilter.FILTER_BY_SAMPLE_INTERNAL_ID,
+        ]
+        metrics = apply_metrics_filter(
+            metrics=self._get_query(table=SampleLaneSequencingMetrics),
+            filter_functions=sample_lane_metrics_filter_functions,
+            sample_internal_id=sample_internal_id,
+            flow_cell_name=flow_cell_name,
+        ).all()
+        total_passing_q30: float = 0
+        for metric in metrics:
+            total_passing_q30 += metric.sample_base_fraction_passing_q30
+
+        return total_passing_q30 / len(metrics) if total_passing_q30 else 0
+
+    def get_number_of_reads_for_flow_cell_from_sample_lane_metrics(
+        self, flow_cell_name: str
+    ) -> int:
+        """Get number of reads for flow cell from sample lane sequencing metrics."""
+        sample_internal_ids: List[
+            str
+        ] = self.get_samples_internal_ids_on_flow_cell_from_sample_lane_metrics(
+            flow_cell_name=flow_cell_name
+        )
+        reads_count: int = 0
+        for sample_internal_id in sample_internal_ids:
+            reads_count += (
+                self.get_number_of_reads_for_sample_on_flow_cell_from_sample_lane_metrics(
+                    sample_internal_id=sample_internal_id, flow_cell_name=flow_cell_name
+                )
+            )
+        return reads_count if reads_count else 0
```

Comment:
We only need one database interaction here right? Fetch the sequencing_metrics for the flow cell and then sum the reads?

Reply:
yes, I simplified these functions by a lot now.